tests/integration/get-images.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# This script checks if container images needed for tests (currently
+# busybox and Debian 10 "Buster") are available locally, and downloads
+# them to testdata directory if not. Its output is suitable for use
+# from shell via eval (see helpers.bash for usage example).
+#
+# XXX: Latest available images are fetched. Theoretically,
+# this can bring some instability in case of a broken image.
+# In this case, images will need to be pinned to a checksum
+# on a per-image and per-architecture basis.
+
+set -e -u -o pipefail
+
+# Root directory of integration tests.
+INTEGRATION_ROOT=$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")
+# Test data path.
+TESTDATA="${INTEGRATION_ROOT}/testdata"
+# Sanity check: $TESTDATA directory must exist.
+if [ ! -d "$TESTDATA" ]; then
+	echo "Bad TESTDATA directory: $TESTDATA. Aborting" >&2
+	exit 1
+fi
+
+function get() {
+	local dest="$1" url="$2"
+
+	[ -e "$dest" ] && return
+
+	# Sanity check: $TESTDATA directory must be writable.
+	if [ ! -w "$TESTDATA" ]; then
+		echo "TESTDATA directory ($TESTDATA) not writable. Aborting" >&2
+		exit 1
+	fi
+
+	curl -o "$dest" -fsSL "$url"
+}
+
+arch=$(go env GOARCH)
+# Convert from GOARCH to whatever the URLs below are using.
+case $arch in
+arm64)
+	arch=arm64v8
+	;;
+esac
+
+# busybox
+BUSYBOX_IMAGE="$TESTDATA/busybox-${arch}.tar.xz"
+get "$BUSYBOX_IMAGE" \
+	"https://github.com/docker-library/busybox/raw/dist-${arch}/stable/glibc/busybox.tar.xz"
+echo "BUSYBOX_IMAGE=$BUSYBOX_IMAGE"
+
+# debian
+DEBIAN_IMAGE="$TESTDATA/debian-${arch}.tar.xz"
+get "$DEBIAN_IMAGE" \
+	"https://github.com/debuerreotype/docker-debian-artifacts/raw/dist-${arch}/buster/rootfs.tar.xz"
```

Comment:
I still think it's a bit dangerous to rely on me not changing these paths again (and would suggest instead embedding an explicit commit reference), but I'm not the one who'll have a failing CI if it changes again so I guess it's your choice? :innocent:

(I certainly can't promise these won't change again. :see_no_evil:)

You might also want to consider grabbing `buster/slim/rootfs.tar.xz` instead if you just need a "basic Debian environment" and aren't too picky about what's in it (since it'll be a smaller download: ~14.6 MB vs ~28.7 MB).

Reply:
Nice, let's try it

Reply:
works!